osu.Game/Rulesets/Replays/FramedAutoGenerator.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+#nullable enable
+
+using System.Collections.Generic;
+using System.Linq;
+using osu.Game.Beatmaps;
+using osu.Game.Replays;
+
+namespace osu.Game.Rulesets.Replays
+{
+    public abstract class FramedAutoGenerator<TFrame> : AutoGenerator
```

Comment:
Don't really get why this is separated out from `AutoGenerator`. `AutoGenerator`'s responsibility is to generate replays. Replays can only be manipulated by adding frames. Therefore there is no reason to *not* want to use `FramedAutoGenerator<T>` (maybe besides the generic spec), because `AutoGenerator` on its own is pretty much useless.

If you implement `AutoGenerator`, you have to either reimplement the entirety of the framed one anyways, or disregard frame order altogether and implement it on your own, in the wrong way. This all just seems really baroque for what the end goal is.

I'd just fuse this with `AutoGenerator` to get an `AutoGenerator<T>`. If the generic becomes a problem (I think there's a constant there somewhere which might make it a problem) then they can live next to each other in one file, at worst.

Reply:
I renamed `FramedAutoGenerator<T>` to `AutoGenerator<T>`. My thought was it is like `FramedReplayInputHandler<T>` vs `ReplayInputHandler`, but I guess that one is also kind of weird at the current state.

I left `AutoGenerator` as is, to keep this PR minimal.